Druva c2c (#1424)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/druva-source.md

[here](https://developer.druva.com/docs/event-apis).

## Prerequisites

Before you begin setting up Druva source, you must meet the following requirements:
* An administrator account is required for **Druva inSync** Cloud.
* Enable the **Export Events** option. To know the steps, see the [Configure Druva inSync](#configure-inSync-to-export-events) section.
* Obtain access tokens by [authentication](#authentication).

### Configure Druva inSync to export events

Only an Druva inSync Cloud administrator can enable the option to export events and define the Events API settings. To get the event logs, follow the steps to enable the **Export Events**:
1. Log in to the Druva inSync Management Console with [inSync Cloud](https://login.druva.com/) account or [inSync GovCloud](https://loginfederal.druva.com/) account.
2. On the [inSync Management Console](https://docs.druva.com/Endpoints/030_Set_up_inSync_for_Endpoints/010_Initial_Configuration/010_Sign_in_to_inSync_Management_Console/Sign_in_to_inSync_Management_Console) menu bar, click **Settings** (wheel icon) > Settings. The Settings page appears.
3. Click the **inSync APIs** tab.

> 4. In the Events API settings area, click **Edit**. The Edit Events API Settings window appears.<br/><img src={useBaseUrl('img/send-data/events-api-settings.png')} alt="events-api-settings.png" width="450"/>

Reviewer: Broken image. It looks like you accidentally put the images in the docs folder  (`/docs/send-data`). Please move them to `/static/img/send-data`.

When you delete the Source, it is placed in a **Stopping** state. When it has successfully stopped, it is deleted from your Hosted Collector.

On the Collection page, the [Health](/docs/manage/health-events#collection-page) and Status for Sources is displayed. Use [Health Events](/docs/manage/health-events) to investigate issues with collection. <br/>![C2C error icon on collection page.png](/img/send-data/C2C-error-icon-on-collection-page.png)

Hover your mouse over the status icon to view a tooltip with a count of the detected errors and warnings.<br/> ![hover c2c error.png](/img/send-data/hover-c2c-error.png)

## Setup and configuration Source

When you create a Druva Source, you add it to a Hosted Collector. Before creating the Source, identify the Hosted Collector you want to use or create a new Hosted Collector. For instructions, see [Configure a Hosted Collector](/docs/send-data/hosted-collectors/configure-hosted-collector.md).

**To configure a Druva Source**

1. In Sumo Logic, select **Manage Data** > **Collection** > **Collection**.
1. On the **Collectors** page, click **Add Source** next to a Hosted Collector.

> 1. Select **Druva**. <br/><img src={useBaseUrl('img/send-data/druva-icon.png')} alt="druva-icon.png" width="200"/>

Reviewer: Broken image. It looks like image is in the docs folder  (`/docs/send-data`). Please move it to `/static/img/send-data`.

1. **Name**. Enter a Name to display for the Source in the Sumo Logic web application.
1. **Description**. (Optional)
1. **Source Category**.(Optional) Enter any string to tag the output collected from the Source. Category metadata is stored in a searchable field called `_sourceCategory`.

> 1. 1. Fields. (Optional) Click **+Add** to ad additional fields; each field needs a name (key) and value.

Reviewer: ```suggestion
1. Fields. (Optional) Click **+Add** to add additional fields; each field needs a name (key) and value.
```

   * ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists and is enabled in the Fields table schema.
   * ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist, or is disabled, in the Fields table schema. In this case, an option to automatically add or enable the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo that does not exist in the Fields schema or is disabled it is ignored, known as dropped.

> 1. **Base URL**. Choose the base url from the drop-down menu as per your requirements — [Data Governance Cloud](https://apis.druva.com) and [Data GovernanceGov Cloud](https://govcloudapis.druva.com).<br/><img src={useBaseUrl('img/send-data/base-url.png')} alt="base-url.png" width="400"/>

Reviewer: Broken image. It looks like you accidentally put the images in the docs folder  (`/docs/send-data`). Please move them to `/static/img/send-data`.

> 8. **Base URL**. Choose the base url from the drop-down menu as per your requirements — [Data Governance Cloud](https://apis.druva.com) and [Data GovernanceGov Cloud](https://govcloudapis.druva.com).<br/><img src={useBaseUrl('img/send-data/base-url.png')} alt="base-url.png" width="400"/>
> 9. **Client ID**. Enter your Client ID generated from your [Druva application](https://developer.druva.com/docs/authentication).

Reviewer: ```suggestion
1. **Base URL**. Choose the base url from the drop-down menu as per your requirements — [Data Governance Cloud](https://apis.druva.com) and [Data GovernanceGov Cloud](https://govcloudapis.druva.com).<br/><img src={useBaseUrl('img/send-data/base-url.png')} alt="base-url.png" width="400"/>
1. **Client ID**. Enter your Client ID generated from your [Druva application](https://developer.druva.com/docs/authentication).
```

> 1. **Secret Key**. Enter your Client secret key generated within your [Druva Application](https://developer.druva.com/docs/authentication).<br/><img src={useBaseUrl('img/send-data/druva-config.png')} alt="druva-config.png" width="600"/>

Reviewer: Broken image. It looks like you accidentally put the images in the docs folder  (`/docs/send-data`). Please move them to `/static/img/send-data`.

Author: That's correct. I have fixed it. Thank you, Kim.

| Parameter | Type | Required | Description | Access |
|:--|:--|:--|:--|:--|
| `config` | JSON Object  | Yes | Contains the [configuration-parameters](#config-parameters) of the Source. | na |
| `schemaRef` | JSON Object  | Yes | Use `{"type":"Druva"}` for Druva Source. | not modifiable |
| `sourceType` | String | Yes | Use `Universal` for Druva Source. | not modifiable |

### Config Parameters

| Parameter | Type | Required | Description | Access |
|:---|:---|:---|:---|:---|
| `name` | String | Yes | Type the desired name of the Source and it must be unique per Collector. This value is assigned to the `metadata field _source`.  | modifiable |
| `description` | String  | No | Type the description of the Source. | modifiable |
| `category` | String | No | Type the category of the source. This value is assigned to the metadata field `_sourceCategory`. | modifiable |
| `fields` | JSON Object | No | JSON map of key-value fields (metadata) to apply to the Collector or Source. Use the boolean field `_siemForward` to enable forwarding to SIEM. | modifiable |

> | `baseURL` | String | Yes | The base URL from which the customer wants to retrieve event data.. | modifiable |

Reviewer: ```suggestion
| `baseURL` | String | Yes | The base URL from which the customer wants to retrieve event data. | modifiable |
```

| `clientID` | String | Yes | Client ID key of an application. | modifiable |

> | `secretKey` | String | Secret key of an application | modifiable |

Reviewer: ```suggestion
| `secretKey` | String | Secret key of an application. | modifiable |
```
